drivers/rilmodem/sim.c

```diff
@@ -661,39 +715,56 @@ static void sim_status_cb(struct ril_msg *message, gpointer user_data)
 		ofono_sim_inserted_notify(sim, FALSE);
 	else
 		ofono_error("%s: bad SIM state (%u)",
-				__func__, status->card_state);
-
-	/* TODO(CDMA): need some kind of logic to set the correct app_index */
-	search_index = status->gsm_umts_index;
-
-	/*
-	 * We cache the current password state. Ideally this should be done by
-	 * issuing a GET_SIM_STATUS request from ril_query_passwd_state, which
-	 * is called by the core after sending a password, but unfortunately the
-	 * response to GET_SIM_STATUS is not reliable in mako when sent just
-	 * after sending the password. Some time is needed before the modem
-	 * refreshes its internal state, and when it does it sends a
-	 * SIM_STATUS_CHANGED event. In that moment we retrieve the status and
-	 * this function is executed. We call __ofono_sim_recheck_pin as it is
-	 * the only way to indicate the core to call query_passwd_state again.
-	 * An option that can be explored in the future is wait before invoking
-	 * core callback for send_passwd until we know the real password state.
-	 */
-	if (status->card_state == RIL_CARDSTATE_PRESENT
-			&& search_index < status->num_apps) {
-		struct reply_sim_app *app = status->apps[search_index];
-
-		if (app->app_type != RIL_APPTYPE_UNKNOWN) {
-			configure_active_app(sd, app, search_index);
-			DBG("passwd_state: %d", sd->passwd_state);
+			    __func__, status->card_state);
 
+	if (status->card_state == RIL_CARDSTATE_PRESENT) {
+		/* TODO(CDMA): need some kind of logic to set the correct app_index */
+		search_index = status->gsm_umts_index;
+		if (search_index > status->num_apps &&
+				sd->vendor == OFONO_RIL_VENDOR_QCOM_MSIM) {
 			/*
-			 * Note: There doesn't seem to be any other way to force
-			 * the core SIM code to recheck the PIN. This call
-			 * causes the core to call the this atom's
-			 * query_passwd() function.
+			 * On QCOM's multi SIM device, all index will be -1
+			 * (but will wrap up to UINT_MAX because index is uint)
+			 * until we send RIL_REQUEST_SET_UICC_SUBSCRIPTION.
+			 * So, we need to figure out which app to use
+			 * and send that request out.
 			 */
-			__ofono_sim_recheck_pin(sim);
+			search_index = sim_select_uicc_subscription(sim, status);
```

Comment:
It's actually only one character over 80, as the trailing line breaks aren't counted.

Take a look at the doc/coding-style.txt, as it has some examples of how this could be corrected.

The simplest would be to just wrap the 2nd parameter:

        		search_index = sim_select_uicc_subscription(sim,
                                                                                                        status);

Note - all tabs are used, this puts the wrapped link just at the 80 char limit.

Or, you could wrap after the "=", like this:

                        search_index =
                                sim_select_uicc_subscription(sim, status);

But I think the first is preferable.